qiskit/quantum_info/operators/symplectic/clifford_circuits.py

```diff
@@ -91,21 +125,13 @@ def _append_circuit(clifford, circuit, qargs=None):
     # are a single qubit Clifford gate rather than raise an exception.
     if gate.definition is None:
         raise QiskitError(f"Cannot apply Instruction: {gate.name}")
-    if not isinstance(gate.definition, QuantumCircuit):
+    if isinstance(gate.definition, QuantumCircuit):
```

Comment:
This is guaranteed by the type `QuantumCircuit` - if some object is returning something else from `definition` it's the object at fault, not this function.  You should be fine to skip the error check here.

Reply:
I've checked the type of `gate.definition` expecting it should be `Optional[QuantumCircuit]`, and I'm surprised it seems undefined, at lease, I couldn't find any document describing it.
Adding the type definition to `gate.definition` may be beyond the scope of this PR, I'm leaning to leave those lines as they are and leave a comment for now. What do you think?

Reply:
We might not have any specific documentation, but it should be guaranteed throughout the library - there are various parts of the transpiler that look at it, and it's only allowed to be `None` or `QuantumCircuit`.  Do you know of anywhere that violates that?

Reply:
OK. Yeah, I don't know anywhere it is violated. I'll remove the check to be practical.

Reply:
Done in [e0ed50c](https://github.com/Qiskit/qiskit-terra/pull/8585/commits/e0ed50c23794a008bcc43fbfa9823df3cc8ac025)